analyses/EvDisp.C

```diff
@@ -0,0 +1,628 @@
+      if(event_eventNumber != evt){
+        continue;
+      }else{
+        cout << "[EvDisp::Loop] processing event  : "<< evt << endl;
+        fill();
+        break;
+      }
+      cout << "[EvDisp::Loop] event  : "<< evt << "not found"<< endl;
+    }else if(evt == 0 || evt < -1){
+      cout << "[EvDisp::Loop] Invalid evt number"<< endl;
+      break;
+    }else{
+      if(!runOnlyIfOneEmptyFlag) cout << "[EvDisp::Loop] processing : "<< jentry << " entry" << endl;
+      else if(jentry%1000 == 0)  cout << "[EvDisp::Loop] processing : "<< jentry << " entry" << endl;
+
+      fill();
+
+      TString continueFlag = "n";
+      if(askContinueFlag){
+        do{
+          if(continueFlag != "n") cout<<"[EvDisp::Loop] Invalid choice"<<endl;
+          cout<<"[EvDisp::Loop] Do you want to go to the next event? y/n(exit)"<<endl;
+          cin>>continueFlag;
+        }while(continueFlag != "y" && continueFlag != "n");
+      }else{
+        continueFlag = "y";
+      }
+
+      
+
+      if(continueFlag == "n") break;
+
+    }
+  }
+
+  cout<<"[EvDisp::Loop] end"<<endl;
+  endJob();
+
+}
+
+void EvDisp::Loop(Long64_t evt)
+{
+  Loop(-1, -1, evt);
+}
+
+void EvDisp::LoopEntry(Long64_t entry)
+{
+  cout << "[EvDisp::Loop] processing entry : "<< entry;
+  Loop(entry, entry+1);
+}
+
+void EvDisp::Loop()
+{
+  Loop(-1,-1,-1);
+}
+
+void EvDisp::book()
+{
+  cout<<"[EvDisp::book] start"<<endl;
+
+  gSystem->Exec("mkdir -p ./evDispPlots");
+
+  zSL1 = computeY(2.5);   // middle of SL1
+  zSL3 = computeY(10.5);  // middle of SL3
+
+  vector<float> xStruct[4], yStruct[4];
+  vector<float> exStruct[4], eyStruct[4];
+
+  // LOOP SL 1-3
+  for(unsigned int iMB=1; iMB<=4; iMB++){
+    for(unsigned int iWire=1; iWire<=nWiresMaxPhi[iMB-1]; iWire++) {
+      for(unsigned int iLayer=1; iLayer<=12; iLayer++){
+        if((iLayer > 4) && (iLayer < 9)) continue;
+        xStruct[iMB-1].push_back(computeX(iWire,iLayer,iMB));
+        yStruct[iMB-1].push_back(computeY(iLayer));
+        exStruct[iMB-1].push_back(cellSizeX/2);
+        eyStruct[iMB-1].push_back(cellSizeY/2);
+      }
+    }
+  }
+
+  // LOOP SL 2
+  for(unsigned int iMB=1; iMB<=3; iMB++){
+    for(unsigned int iWire=1; iWire<=nWiresMaxEta; iWire++) {
+      for(unsigned int iLayer=5; iLayer<=8; iLayer++){
+        xStruct[iMB-1].push_back(computeX(iWire,iLayer,iMB));
+        yStruct[iMB-1].push_back(computeY(iLayer));
+        exStruct[iMB-1].push_back(cellSizeX/2);
+        eyStruct[iMB-1].push_back(cellSizeY/2);
+      }
+    }
+  }
+
+  graphStruct = new TGraphErrors*[4];
+  for(int i=0;i<4;i++){
+    graphStruct[i] = new TGraphErrors(xStruct[i].size(),&xStruct[i][0],&yStruct[i][0],&exStruct[i][0],&eyStruct[i][0]); 
+    graphStruct[i]->SetMarkerStyle(1);
+    graphStruct[i]->GetXaxis()->SetLimits(0.,410.);
+    graphStruct[i]->SetName("graphStruct");
+  }
+
+  cout<<"[EvDisp::book] end"<<endl;
+
+}
+
+void EvDisp::fill()
+{
+
+  // STRUCT DRAWING
+  auto c1 = new TCanvas("c1","c1",1500,1000);
```

Comment:
Please reduce the 1500 - 1000 canvas size to 1200 - 800.
A bigger canvas causes problems, at least on MACs.

Also, if the canvas could be created at the beginning (e.g. in book) and re-filled every event, instead of being deleted and recreated, this would make the focus of the prompt remain in the root prompt and make use easier.
This last feature could be implemented in a future version though.

Reply:
Both requests were implemented and things appear to be working fine